program_intent_engagement/apps/api/v1/tests/test_views.py

```diff
@@ -298,3 +299,46 @@ def test_multiple_intents(self):
 
         self.assertIn(self.p1_intent_maybe.certainty, actual_certainties)
         self.assertIn(p2_intent_certain.certainty, actual_certainties)
+
+    def test_multiple_intents_same(self):
+        """
+        Test that when user has multiple program intents with the same certainty,
+        you get the most recent intent.
+        """
+
+        p2_intent_certain_recent = ProgramIntent.objects.create(
+            lms_user_id=self.user.lms_user_id,
+            program_uuid=self.program_id2,
+            reason="a reason",
+            certainty="CERTAIN_NO",
+            effective_timestamp=datetime.datetime(2021, 9, 17),
+        )
+        ProgramIntent.objects.create(
+            lms_user_id=self.user.lms_user_id,
+            program_uuid=self.program_id2,
+            reason="a reason",
+            certainty="CERTAIN_YES",
+            effective_timestamp=datetime.datetime(2021, 7, 17),
+        )
+        ProgramIntent.objects.create(
+            lms_user_id=self.user.lms_user_id,
+            program_uuid=self.program_id2,
+            reason="a reason",
+            certainty="MAYBE",
+            effective_timestamp=datetime.datetime(2021, 9, 17),
+        )
+
+        response = MostRecentAndCertainIntentsView.grab_best_intents(
+            ProgramIntent.objects
+        )
+        responses_list = list(response.values())
+
+        self.assertEqual(len(response.values()), 2)
+
+        actual_certainties = [
+            responses_list[0]["certainty"],
+            responses_list[1]["certainty"],
+        ]
+
+        self.assertIn(self.p1_intent_maybe.certainty, actual_certainties)
+        self.assertIn(p2_intent_certain_recent.certainty, actual_certainties)
```

Comment:
testing nit - I'd actually test the reasons because you can give them all a unique reason that doesn't mean anything else, where the certainties mean something

Reply:
Good idea! Updated.